src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeRecordsAndCompactCtors.java

```diff
@@ -24,7 +24,7 @@
 
 public class InputIllegalTypeRecordsAndCompactCtors {
     record MyTestRecord
-            (LinkedHashMap<Integer, Integer> linkedHashMap) { // violation
+            (LinkedHashMap<Integer, Integer> linkedHashMap) { // OK
 
```

Comment:
It should be violation

Reply:
I think the condition is not working with these violations.

Reply:
We need to keep violation. Code should be updated.

Reply:
Can we use LinkedHashUp as a Parameter?